src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LoweredDynamicOperationFactory.cs

```diff
@@ -660,7 +662,26 @@ internal static RefKind GetReceiverRefKind(BoundExpression loweredReceiver)
                 delegateInvoke,
                 callSiteArguments);
 
-            return new LoweredDynamicOperation(_factory, siteInitialization, siteInvocation, resultType);
+            return new LoweredDynamicOperation(_factory, siteInitialization, siteInvocation, resultType, temps);
+        }
+
+        /// <summary>
+        /// If there are any discards in the arguments, create locals for each, updates the arguments and
+        /// returns the symbols that were created.
+        /// Returns default if no discards found.
+        /// </summary>
+        private ImmutableArray<LocalSymbol> MakeTempsForDiscardArguments(ref ImmutableArray<BoundExpression> loweredArguments)
+        {
+            int discardCount = loweredArguments.Count(a => a.Kind == BoundKind.DiscardedExpression);
+
+            if (discardCount == 0)
+            {
+                return default(ImmutableArray<LocalSymbol>);
```

Comment:
>return default(ImmutableArray<LocalSymbol>); [](start = 16, length = 44)

Why not empty? It feels like if there are no locals, the result should be empty.